GitUI/CommitInfo/CommitInfo.cs

```diff
@@ -183,8 +183,7 @@ private void ReloadCommitInfo()
             if (AppSettings.ShowGpgInformation)
             {
                 /* Setup GpgController */
-                _gpgController.Revision = _revision;
-                _gpgController.Module = Module;
+                _gpgController = new GitGpgController(Module, _revision);
```

Comment:
Module is returned from IGitUICommandsSource and it changes when the current repository changes. We can not pass it  to the global _gpgController field (it is global because CommitInfo is global). Either GitGpgController must depend on IGitUICommandsSource or GitGpgController has to be created before each use.

Reply:
In the proposed change the controller is instantiated for each revision selected

Reply:
Right, didn't notice that, I thought it was created in the constructor. I take it back.

Reply:
I did the same changes here and it works as expected. I'm going to merge these changes into the branch, thanks.

Reply:
Just for completeness I would assign null to the `_gpgController ` at the beginning of the `ReloadCommitInfo` to ensure that a stale instance is referenced nowhere.

Reply:
Agree, good spot

Reply:
you're right, I'm going to add it now.